### ISSUE TITLE
After pod start network temporarily is unavailable

### ISSUE THREAD

**Issue by iLeonidze** [projectcalico/calico#3628]
At the start of the pod, it sometimes has non-working network for the first few seconds.

## Expected Behavior
When the pod starts, the network is immediately completely ready for use.

## Current Behavior
At the moment, some applications immediately crash, so they do not have a network. If you add a delay, everything works.

## Possible Solution
As far as I see, the network interface first has `tentative`, and then it disappears and the network works. Possibly problems with DAD, may be here the rootcause https://unix.stackexchange.com/questions/293629/let-ifconfig-wait-for-ipv6-address-to-not-be-tentative

## Steps to Reproduce (for bugs)
1. Create and apply Daemonset
    ```
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      name: pingtest
      namespace: kube-system
      labels:
        k8s-app: busybox
    spec:
      selector:
        matchLabels:
          name: pingtest
      template:
        metadata:
          labels:
            name: pingtest
        spec:
          tolerations:
          - key: node-role.kubernetes.io/master
            effect: NoSchedule
          containers:
          - name: pingtest
            image: busybox-curl
            command: [ "sh", "-c", "ip a; curl -vk https://kubernetes.default; sleep 10s; ip a; curl -vk https://kubernetes.default" ]
          terminationGracePeriodSeconds: 3
    ```
2. Read logs. I get the following result:
    ```
    1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
        link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
        inet 127.0.0.1/8 scope host lo
           valid_lft forever preferred_lft forever
        inet6 ::1/128 scope host 
           valid_lft forever preferred_lft forever
    3: eth0@if212: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1440 qdisc noqueue 
        link/ether be:99:51:06:f1:f6 brd ff:ff:ff:ff:ff:ff
        inet6 fd02::1626:ef84:3d30/128 scope global tentative 
           valid_lft forever preferred_lft forever
        inet6 fe80::bc99:51ff:fe06:f1f6/64 scope link tentative 
           valid_lft forever preferred_lft forever

    * Expire in 200 ms for 1 (transfer 0x5599dcc23880)
      0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0* Expire in 200 ms for 1 (transfer 0x5599dcc23880)
    * Could not resolve host: kubernetes.default
    * Expire in 200 ms for 1 (transfer 0x55b2bbd58880)
    * Closing connection 0
    curl: (6) Could not resolve host: kubernetes.default

    1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
        link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
        inet 127.0.0.1/8 scope host lo
           valid_lft forever preferred_lft forever
        inet6 ::1/128 scope host 
           valid_lft forever preferred_lft forever
    3: eth0@if212: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1440 qdisc noqueue 
        link/ether be:99:51:06:f1:f6 brd ff:ff:ff:ff:ff:ff
        inet6 fd02::1626:ef84:3d30/128 scope global 
           valid_lft forever preferred_lft forever
        inet6 fe80::bc99:51ff:fe06:f1f6/64 scope link 
           valid_lft forever preferred_lft forever

    * TCP_NODELAY set
    * Expire in 200 ms for 4 (transfer 0x5564f127d880)
    * Connected to kubernetes.default (fd03::7d4) port 80 (#0)
    > GET / HTTP/1.1
    > Host: kubernetes.default
    > User-Agent: curl/7.64.0
    > Accept: /
    >
    < HTTP/1.1 403 Forbidden
    ```

## Context
I am running a thirdparty software that requires immediate access to the network. Adding a sleep does not seem to me the correct solution.

## Your Environment
* Calico version v3.10.1
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes v1.16.3
* Operating System and version: Red Hat Enterprise Linux Server release 7.6 (Maipo)
* Calico config: [calico.yaml](https://github.com/projectcalico/calico/files/4722443/calico.yaml.txt)

**Comment by fasaxc** [comment]
I thought we'd fixed an issue where the IPv6 address remains tentative a while ago.  Might be worth upgrading to the current supported version to see if this is already fixed.  v3.14 is the current supported version.

**Comment by iLeonidze** [comment]
Hi, we can’t upgrade to the new version, so we can only explore and provide logs by what we have.

I created a script that every 10 milliseconds shows `ip a` on the interface. I launched the container  with this entrypoint and here are the logs that turned out:
```
[root@master-1 kubernetes]# kubectl logs pingtest-bjzvj -n kube-system | head -n 400
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global tentative 
    inet6 fd02::1626:ef84:3d2e/128 scope global 
    inet6 fd02::1626:ef84:3d2e/128 scope global 
    inet6 fd02::1626:ef84:3d2e/128 scope global 
    inet6 fd02::1626:ef84:3d2e/128 scope global 
    inet6 fd02::1626:ef84:3d2e/128 scope global
```

If we count, we will see that the network began to work after `690` milliseconds (I'm sorry, I can’t print the number of milliseconds since the epoch as a log in the container). 

Now let's look at the Calico `calico-node` pod log:
```
2020-06-18 09:42:43.410 [INFO][54] ipsets.go 356: Finished resync family="inet6" numInconsistenciesFound=0 resyncDuration=2.491771ms
2020-06-18 09:42:43.410 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=2.993589
2020-06-18 09:42:50.037 [INFO][54] int_dataplane.go 630: Linux interface addrs changed. addrs=set.mapSet{} ifaceName="calia35c49471eb"
2020-06-18 09:42:50.037 [INFO][54] int_dataplane.go 864: Received interface addresses update msg=&intdataplane.ifaceAddrsUpdate{Name:"calia35c49471eb", Addrs:set.mapSet{}}
2020-06-18 09:42:50.038 [INFO][54] hostip_mgr.go 84: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"calia35c49471eb", Addrs:set.mapSet{}}
2020-06-18 09:42:50.038 [INFO][54] int_dataplane.go 597: Linux interface state changed. ifaceName="calia35c49471eb" state="up"
2020-06-18 09:42:50.038 [INFO][54] int_dataplane.go 630: Linux interface addrs changed. addrs=set.mapSet{"fe80::ecee:eeff:feee:eeee":set.empty{}} ifaceName="calia35c49471eb"
2020-06-18 09:42:50.039 [INFO][54] hostip_mgr.go 84: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"calia35c49471eb", Addrs:set.mapSet{}}
2020-06-18 09:42:50.039 [INFO][54] int_dataplane.go 864: Received interface addresses update msg=&intdataplane.ifaceAddrsUpdate{Name:"calia35c49471eb", Addrs:set.mapSet{"fe80::ecee:eeff:feee:eeee":set.empty{}}}
2020-06-18 09:42:50.039 [INFO][54] hostip_mgr.go 84: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"calia35c49471eb", Addrs:set.mapSet{"fe80::ecee:eeff:feee:eeee":set.empty{}}}
2020-06-18 09:42:50.039 [INFO][54] hostip_mgr.go 84: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"calia35c49471eb", Addrs:set.mapSet{"fe80::ecee:eeff:feee:eeee":set.empty{}}}
2020-06-18 09:42:50.039 [INFO][54] int_dataplane.go 959: Applying dataplane updates
2020-06-18 09:42:50.039 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=0.24063099999999998
2020-06-18 09:42:50.039 [INFO][54] int_dataplane.go 846: Received interface update msg=&intdataplane.ifaceUpdate{Name:"calia35c49471eb", State:"up"}
2020-06-18 09:42:50.039 [INFO][54] int_dataplane.go 959: Applying dataplane updates
2020-06-18 09:42:50.040 [INFO][54] endpoint_mgr.go 331: Workload interface came up, marking for reconfiguration. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.040 [INFO][54] endpoint_mgr.go 948: Applying /proc/sys configuration to interface. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.040 [INFO][54] endpoint_mgr.go 331: Workload interface came up, marking for reconfiguration. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.040 [INFO][54] endpoint_mgr.go 948: Applying /proc/sys configuration to interface. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.041 [INFO][54] route_table.go 556: Syncing routes: found unexpected route; ignoring due to grace period. dest=fd02::1626:ef84:3d2e/128 ifaceName="calia35c49471eb" ipVersion=0x6
2020-06-18 09:42:50.041 [INFO][54] route_table.go 561: Syncing routes: removing old route. dest=fe80::/64 ifaceName="calia35c49471eb" ipVersion=0x6 routeProblems=[]string{"incorrect protocol"}
2020-06-18 09:42:50.041 [INFO][54] route_table.go 364: Interface in cleanup grace period, will retry after. ifaceName="calia35c49471eb" ipVersion=0x6
2020-06-18 09:42:50.041 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=1.696461
2020-06-18 09:42:50.082 [INFO][54] calc_graph.go 392: Local endpoint updated id=WorkloadEndpoint(node=app-node-2, orchestrator=k8s, workload=kube-system/pingtest-bjzvj, name=eth0)
2020-06-18 09:42:50.082 [INFO][54] int_dataplane.go 831: Received *proto.ActiveProfileUpdate update from calculation graph msg=id:<name:"ksa.kube-system.pingtest-sa" > profile:<> 
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pri-_DrANhs9NJgtQWDtTHW" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 830: Invalidating dataplane cache ipVersion=0x4 reason="chain update" table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pro-_DrANhs9NJgtQWDtTHW" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pri-_DrANhs9NJgtQWDtTHW" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 830: Invalidating dataplane cache ipVersion=0x6 reason="chain update" table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pro-_DrANhs9NJgtQWDtTHW" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.083 [INFO][54] int_dataplane.go 831: Received *proto.ActiveProfileUpdate update from calculation graph msg=id:<name:"kns.kube-system" > profile:<inbound_rules:<action:"allow" rule_id:"YlN-KJA93B7Fe2ih" > outbound_rules:<action:"allow" rule_id:"GUM3x2IgYhIM5cBY" > > 
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pri-kns.kube-system" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pro-kns.kube-system" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pri-kns.kube-system" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-pro-kns.kube-system" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.083 [INFO][54] int_dataplane.go 831: Received *proto.WorkloadEndpointUpdate update from calculation graph msg=id:<orchestrator_id:"k8s" workload_id:"kube-system/pingtest-bjzvj" endpoint_id:"eth0" > endpoint:<state:"active" name:"calia35c49471eb" profile_ids:"kns.kube-system" profile_ids:"ksa.kube-system.pingtest-sa" ipv6_nets:"fd02::1626:ef84:3d2e/128" > 
2020-06-18 09:42:50.083 [INFO][54] int_dataplane.go 959: Applying dataplane updates
2020-06-18 09:42:50.083 [INFO][54] endpoint_mgr.go 497: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-tw-calia35c49471eb" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.083 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-fw-calia35c49471eb" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.083 [INFO][54] endpoint_mgr.go 524: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-from-wl-dispatch-a" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-from-wl-dispatch" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-to-wl-dispatch-a" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-to-wl-dispatch" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.084 [INFO][54] endpoint_mgr.go 948: Applying /proc/sys configuration to interface. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.084 [INFO][54] endpoint_mgr.go 432: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.084 [INFO][54] status_combiner.go 58: Storing endpoint status update ipVersion=0x4 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.084 [INFO][54] endpoint_mgr.go 497: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-tw-calia35c49471eb" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-fw-calia35c49471eb" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.084 [INFO][54] endpoint_mgr.go 524: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-from-wl-dispatch-a" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-from-wl-dispatch" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.084 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-to-wl-dispatch-a" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.085 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-to-wl-dispatch" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.085 [INFO][54] endpoint_mgr.go 948: Applying /proc/sys configuration to interface. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.085 [INFO][54] endpoint_mgr.go 432: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.085 [INFO][54] status_combiner.go 58: Storing endpoint status update ipVersion=0x6 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.091 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x4 table="filter"
2020-06-18 09:42:50.093 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x6 table="filter"
2020-06-18 09:42:50.157 [INFO][54] status_combiner.go 81: Endpoint up for at least one IP version id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"} ipVersion=0x4 status="up"
2020-06-18 09:42:50.157 [INFO][54] status_combiner.go 98: Reporting combined status. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"} status="up"
2020-06-18 09:42:50.157 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=74.057322
2020-06-18 09:42:50.784 [INFO][54] calc_graph.go 392: Local endpoint updated id=WorkloadEndpoint(node=app-node-2, orchestrator=k8s, workload=kube-system/pingtest-bjzvj, name=eth0)
2020-06-18 09:42:50.784 [INFO][54] int_dataplane.go 831: Received *proto.WorkloadEndpointUpdate update from calculation graph msg=id:<orchestrator_id:"k8s" workload_id:"kube-system/pingtest-bjzvj" endpoint_id:"eth0" > endpoint:<state:"active" name:"calia35c49471eb" profile_ids:"kns.kube-system" profile_ids:"ksa.kube-system.pingtest-sa" ipv6_nets:"fd02::1626:ef84:3d2e/128" > 
2020-06-18 09:42:50.784 [INFO][54] int_dataplane.go 959: Applying dataplane updates
2020-06-18 09:42:50.784 [INFO][54] endpoint_mgr.go 497: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.784 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-tw-calia35c49471eb" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.784 [INFO][54] table.go 830: Invalidating dataplane cache ipVersion=0x4 reason="chain update" table="filter"
2020-06-18 09:42:50.784 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-fw-calia35c49471eb" ipVersion=0x4 table="filter"
2020-06-18 09:42:50.784 [INFO][54] endpoint_mgr.go 524: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.785 [INFO][54] endpoint_mgr.go 948: Applying /proc/sys configuration to interface. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.785 [INFO][54] endpoint_mgr.go 432: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.785 [INFO][54] status_combiner.go 58: Storing endpoint status update ipVersion=0x4 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.785 [INFO][54] endpoint_mgr.go 497: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.785 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-tw-calia35c49471eb" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.785 [INFO][54] table.go 830: Invalidating dataplane cache ipVersion=0x6 reason="chain update" table="filter"
2020-06-18 09:42:50.785 [INFO][54] table.go 475: Queueing update of chain. chainName="cali-fw-calia35c49471eb" ipVersion=0x6 table="filter"
2020-06-18 09:42:50.785 [INFO][54] endpoint_mgr.go 524: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.786 [INFO][54] endpoint_mgr.go 948: Applying /proc/sys configuration to interface. ifaceName="calia35c49471eb"
2020-06-18 09:42:50.786 [INFO][54] endpoint_mgr.go 432: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.786 [INFO][54] status_combiner.go 58: Storing endpoint status update ipVersion=0x6 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"}
2020-06-18 09:42:50.789 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x4 table="filter"
2020-06-18 09:42:50.793 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x6 table="filter"
2020-06-18 09:42:50.803 [INFO][54] status_combiner.go 81: Endpoint up for at least one IP version id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"} ipVersion=0x6 status="up"
2020-06-18 09:42:50.803 [INFO][54] status_combiner.go 98: Reporting combined status. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"} status="up"
2020-06-18 09:42:50.803 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=19.375199
2020-06-18 09:42:51.123 [INFO][54] int_dataplane.go 959: Applying dataplane updates
2020-06-18 09:42:51.124 [INFO][54] table.go 830: Invalidating dataplane cache ipVersion=0x6 reason="post update" table="filter"
2020-06-18 09:42:51.126 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x6 table="filter"
2020-06-18 09:42:51.136 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=12.151723
2020-06-18 09:42:51.169 [INFO][54] int_dataplane.go 959: Applying dataplane updates
2020-06-18 09:42:51.170 [INFO][54] table.go 830: Invalidating dataplane cache ipVersion=0x4 reason="post update" table="filter"
2020-06-18 09:42:51.171 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x4 table="filter"
2020-06-18 09:42:51.180 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=10.300689
2020-06-18 09:42:52.001 [INFO][54] iface_monitor.go 173: Netlink address update. addr="fe80::ecee:eeff:feee:eeee" exists=true ifIndex=311
2020-06-18 09:42:52.116 [INFO][54] int_dataplane.go 959: Applying dataplane updates
```

You may find that first line about test pod appeared at `2020-06-18 09:42:50.082`. Let's add `690` milliseconds and see that part of log:
```
2020-06-18 09:42:50.093 [INFO][54] table.go 518: Loading current iptables state and checking it is correct. ipVersion=0x6 table="filter"
2020-06-18 09:42:50.157 [INFO][54] status_combiner.go 81: Endpoint up for at least one IP version id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"} ipVersion=0x4 status="up"
2020-06-18 09:42:50.157 [INFO][54] status_combiner.go 98: Reporting combined status. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system/pingtest-bjzvj", EndpointId:"eth0"} status="up"
2020-06-18 09:42:50.157 [INFO][54] int_dataplane.go 973: Finished applying updates to dataplane. msecToApply=74.057322
2020-06-18 09:42:50.784 [INFO][54] calc_graph.go 392: Local endpoint updated id=WorkloadEndpoint(node=app-node-2, orchestrator=k8s, workload=kube-system/pingtest-bjzvj, name=eth0)
```

Judging by the log, something really happened these `690` seconds.
@fasaxc Can you give an explanation of what Calico is doing at this point in so much time?

I also noticed that time varies. For example, I made a DaemonSet for several nodes and looked at how long this lag took on each of the pods - here is the result:
```
1,200s
1,000s
0,900s
0,900s
1,400s
0,800s
1,200s
0,300s
0,900s
```
Average lag time is `~0,95` seconds. I assume that it is too long for the DAD IPv6 address matching, isn’t it?

**Comment by fasaxc** [comment]
Just checking, are you using Calico CNI (as opposed to say, Calico with Google/Amazon CNI)?

**Comment by fasaxc** [comment]
I put up a PR to disable DAD; I think that's a good idea whether it's causing this issue or not: https://github.com/projectcalico/cni-plugin/pull/895

**Comment by fasaxc** [comment]
I believe this was fixed in v3.16.0
